docs/halibut-protocol.md

```diff
@@ -0,0 +1,55 @@
+# Halibut Protocol data exchange
```

Comment:
Updated to make mention of the control messages during initiation as well as some hints at what the messages are trying to signal.

Reply:
Starting to look good, although I'm finding the numbered steps with the "Steps 2 to 5 are repeated..." slightly hard to follow. 

I wonder if it might be helpful to split the numbered list into 3 sub-sections to help show the various phases of the connection:
- Identification
- Message exchange
- Ending a connection

Having separate sub-sections might allow more room to expand on some things like the difference between graceful and non-graceful termination of the connection (e.g. do both parties send the `END` message?)

What do you think?

Reply:
I tried to do this, did it work?

Reply:
Yep that reads much better, thanks.